core/src/main/java/hudson/ExtensionList.java

```diff
@@ -418,6 +418,19 @@ private void fireOnChangeListeners() {
     }
 
     /**
+     * Convenience method allowing lookup of the first instance of a given type.
+     * Equivalent to {@code ExtensionList.lookup(Class).get(Class)}.
+     *
+     * @param type The type to look up.
+     * @return the first instance of the given type in its list.
+     *
+     * @since TODO
+     */
+    public static @CheckForNull <U> U lookupFirst(Class<U> type) {
```

Comment:
`@CheckForNull` here more or less defeats the real need for this, which is for cases where you expect that there *must* be one instance. I would rather say something like

```java
ExtensionList<U> all = lookup(type);
if (all.size() != 1) throw new IllegalStateException(…);
return all.get(0);
```

And name it something like `lookUpSingleton`.

See for example `GlobalConfigurationCategory.get`, `Jenkins.doConfigExecutorsSubmit` call to `MasterBuildConfiguration`, `JenkinsLocationConfiguration.get`, `UserDetailsCache.get`, ad nauseam. Basically just Find Usages on `ExtensionList.get(Class)`. If you know that the singleton has been registered, then you expect to be able to load it without null checks. Maybe existing calls in fact just assume that the result is not null, meaning FindBugs would complain if we actually let it.